### PR TITLE
Move alternative logins to events

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: lib/private/legacy/OC_App.php [hunk 1]
```diff
@@ -665,16 +666,30 @@ public static function registerPersonal(string $app, string $page) {
 
 	/**
 	 * @param array $entry
+	 * @depreacted 20.0.0 Use \OCP\Authentication\Events\AlternativeLoginOptionsEvent
 	 */
 	public static function registerLogIn(array $entry) {
```
Comment (nickvergessen) [inline lib/private/legacy/OC_App.php:671]: I'm also fine with dropping this immediately.
I guess apart from the registration app some apps from https://apps.nextcloud.com/?search=login might need adjustment

**Review by ChristophWurst — APPROVED** [review]
Not a fan of mutable events but :elephant:

**Comment by ChristophWurst** [comment]
Let me try something in another PR that helps move things like this to registration but still make it kind of lazy.

**Comment by nickvergessen** [comment]
Replaced by #21850

**Pull request closed without merging**